class.json-api.php

```diff
@@ -616,6 +616,85 @@ function get_avatar_url( $email, $avatar_size = null ) {
 		}
 	}
 
+	/**
+	 * Counts the number of comments on a site, excluding certain comment types.
+	 *
+	 * @param $post_id int Post ID.
+	 * @return array Array of counts, matching the output of https://developer.wordpress.org/reference/functions/get_comment_count/.
+	 */
+	public function wp_count_comments( $post_id ) {
+		global $wpdb;
+		if ( 0 !== $post_id ) {
+			return wp_count_comments( $post_id );
+		}
+
+		$counts = array(
+			'total_comments' => 0,
+			'all'            => 0,
+		);
+
+		/**
+		 * Exclude certain comment types from comment counts in the REST API.
+		 *
+		 * @since 6.9.0
+		 * @module json-api
+		 *
+		 * @param array Array of comment types to exclude (default: 'order_note', 'webhook_delivery', 'review', 'action_log')
+		 */
+		$exclude = apply_filters( 'jetpack_api_exclude_comment_types_count',
+			array( 'order_note', 'webhook_delivery', 'review', 'action_log' )
+		);
+
+		if ( empty( $exclude ) ) {
+			return wp_count_comments( $post_id );
+		}
+
+		$where = 'WHERE comment_type NOT IN (';
```

Comment:
Would it be more readable to do something like this? I have not tested it though, I just thought it seemed easier to read?

```suggestion
		$where = sprintf(
			"WHERE comment_type NOT IN ('%s')",
			esc_html( implode( "','", $exclude ) )
		);
```

I also added some escaping there, which I think is important since we have a filter allowing one to bring their own data here.

Reply:
Good call. That's much better and works. I've replaced `esc_html` here with `esc_sql`, and applied it to each array element instead (otherwise lots of extra slashes are added).